src/specializations/Triangle.jl

```diff
@@ -87,14 +87,40 @@ function integral(
         v = R * (1 - b / (a + b))
         return f(triangle(u, v)) * R / (a + b)^2
     end
-    ∫ = HCubature.hcubature(integrand, _zeros(FP, 2), (FP(1), FP(π / 2)), rule.kwargs...)[1]
+
+    # HCubature doesn't support functions that output Unitful Quantity types
+    # Establish the units that are output by f
+    testpoint_parametriccoord = _zeros(T, 2)
+    integrandunits = Unitful.unit.(integrand(testpoint_parametriccoord))
+    # Create a wrapper that returns only the value component in those units
+    uintegrand(ts) = Unitful.ustrip.(integrandunits, integrand(ts))
+    # Integrate only the unitless values
+    ∫ = HCubature.hcubature(uintegrand, _zeros(FP, 2), (T(1), T(π / 2)), rule.kwargs...)[1]
 
     # Apply a linear domain-correction factor 0.5 ↦ area(triangle)
-    return 2 * Meshes.area(triangle) .* ∫
+    return 2 * Meshes.area(triangle) * integrandunits .* ∫
```

Comment:
This fixes a bug discovered along the way. The `integral(f, ::Triangle, ::HAdaptiveCubature)` method was never updated to include a Unitful integrand wrapper like the general one has, so integrating any function that produces Unitful outputs will throw an error.

Reply:
That's good to know. And the tests didn't fail because they didn't test any unitful integrand and therefore it worked by accident? Maybe we should test with unitful integrands everywhere? I just saw that some integrand in the tests are unitful and most are unitless. I would assume that if uniful integrands work, unitless integrand also work. So maybe that would be something for another PR to avoid missing these bugs?

Reply:
Thanks for finding and fixing this!

Reply:
> That's good to know. And the tests didn't fail because they didn't test any unitful integrand and therefore it worked by accident?

Yeah, the test item for Triangle didn’t have a Unitful integrand, dating to before the update, so it wasn’t caught until benchmarks were failing (because `main` threw an error on this case).

> Maybe we should test with unitful integrands everywhere? I just saw that some integrand in the tests are unitful and most are unitless. I would assume that if uniful integrands work, unitless integrand also work. So maybe that would be something for another PR to avoid missing these bugs?

I have an update to the test system in mind that I’m trying to work out conceptually, that I hope will be more comprehensive, to include testing callable structs as f. 
I’m hoping to get work on that done over the coming weeks.